airflow/decorators/docker.py

```diff
@@ -0,0 +1,159 @@
+        unrolled to multiple XCom values. Dict will unroll to xcom values with keys as keys.
+        Defaults to False.
+    :type multiple_outputs: bool
+    """
+
+    template_fields = ('op_args', 'op_kwargs')
+    template_fields_renderers = {"op_args": "py", "op_kwargs": "py"}
+
+    # since we won't mutate the arguments, we should just do the shallow copy
+    # there are some cases we can't deepcopy the objects (e.g protobuf).
+    shallow_copy_attrs = ('python_callable',)
+
+    @apply_defaults
+    def __init__(
+        self,
+        **kwargs,
+    ) -> None:
+        self.string_args = [1, 2, 1]
+        self._output_filename = ""
+        command = "dummy command"
+        self.pickling_library = pickle
+        super().__init__(
+            command=command, retrieve_output=True, retrieve_output_path="/tmp/script.out", **kwargs
+        )
+
+    def execute(self, context: Dict):
+        with TemporaryDirectory(prefix='venv') as tmp_dir:
+            input_filename = os.path.join(tmp_dir, 'script.in')
+            self._output_filename = os.path.join(tmp_dir, 'script.out')
+            string_args_filename = os.path.join(tmp_dir, 'string_args.txt')
+            script_filename = os.path.join(tmp_dir, 'script.py')
+            self._write_args(input_filename)
+            self._write_string_args(string_args_filename)
+            py_source = self._get_python_source()
+            write_python_script(
+                jinja_context=dict(
+                    op_args=self.op_args,
+                    op_kwargs=self.op_kwargs,
+                    pickling_library=self.pickling_library.__name__,
+                    python_callable=self.python_callable.__name__,
+                    python_callable_source=py_source,
+                ),
+                filename=script_filename,
+            )
+
+            self.environment["PYTHON_SCRIPT"] = _b64_encode_file(script_filename)
+            if self.op_args or self.op_kwargs:
+                self.environment["PYTHON_INPUT"] = _b64_encode_file(input_filename)
+            else:
+                self.environment["PYTHON_INPUT"] = ""
+
+            self.command = (
+                f'bash -cx \'{_generate_decode_command("PYTHON_SCRIPT", "/tmp/script.py")} &&'
+                f'touch /tmp/string_args &&'
+                f'touch /tmp/script.in &&'
+                f'{_generate_decode_command("PYTHON_INPUT", "/tmp/script.in")} &&'
+                f'python /tmp/script.py /tmp/script.in /tmp/script.out /tmp/string_args \''
+            )
+            return super().execute(context)
+
+    def _write_args(self, filename):
+        if self.op_args or self.op_kwargs:
+            with open(filename, 'wb') as file:
+                self.pickling_library.dump({'args': self.op_args, 'kwargs': self.op_kwargs}, file)
+
+    def _write_string_args(self, filename):
+        with open(filename, 'w') as file:
+            file.write('\n'.join(map(str, self.string_args)))
```

Comment:
Two things here:

1. This is duplicating code from VenvOperator, we should fix a way to not need that (make a mixin perhaps?)
2. The whole `self.string_args` is a misfeature of Venv we shouldn't copy over anyway.